sw/device/lib/base/memory.c

```diff
@@ -12,26 +14,109 @@
 #define OT_PREFIX_IF_NOT_RV32(name) ot_##name
 #endif
 
+static size_t compute_num_leading_bytes(uintptr_t left, uintptr_t right,
+                                        size_t len) {
+#ifndef OT_PLATFORM_RV32
+  assert(len == 0 || left != (uintptr_t)NULL);
+#endif
+  if (len < alignof(uint32_t)) {
+    return len;
+  }
+  const size_t left_ahead = misalignment32_of((uintptr_t)left);
+  const size_t right_ahead = misalignment32_of((uintptr_t)right);
+  if (right == (uintptr_t)NULL || left_ahead == right_ahead) {
+    return (4 - left_ahead) & 0x3;
+  }
+  return len;
+}
+
+/**
+ * Compute the bounds of the word-aligned region for the given buffers.
+ *
+ * It's more efficient for our memory functions to operate on `uint32_t` values
+ * than individual bytes, but we can only read `uint32_t` values from aligned
+ * addresses. This function effectively breaks the given buffers into three
+ * consecutive chunks: (1) unaligned leading bytes, (2) aligned words, and (3)
+ * unaligned trailing bytes.
+ *
+ * @param[in] left The memory function's first buffer argument. Cannot be NULL.
+ * @param[in] right The memory function's second buffer argument. May be NULL.
+ * @param[in] len The length in bytes of both `left` and `right.`
+ * @param[out] out_leading_bytes_end The end of the region of leading bytes.
+ * @param[out] out_aligned_words_end The end of the region of aligned words.
+ */
+static void compute_alignment(uintptr_t left, uintptr_t right, size_t len,
```

Comment:
nit: instead of `leading_bytes_end` and `aligned_words_end`, wdyt about `body_offset` and `tail_offset`, respectively? (offset=byte offset from the start of the buffer)

Reply:
Oh, I kind of misread this. Fixed now.